hal/armv7a/pmap.c

```diff
@@ -115,47 +119,38 @@ static const u16 attrMap[] = {
 
 static void _pmap_asidAlloc(pmap_t *pmap)
 {
-	pmap_t *evicted = NULL;
-
-	while (!++pmap_common.asidptr || (evicted = pmap_common.asid_map[pmap_common.asidptr]) != NULL) {
-		if (evicted != NULL) {
-			if ((hal_cpuGetContextId() & 0xff) == pmap_common.asids[evicted->asid_ix])
-				continue;
+	u8 newAsid = pmap_common.asidCnt++;
 
-			evicted->asid_ix = 0;
-			break;
-		}
+	if (newAsid == ASID_INVALID) {
+		newAsid = 1;
+		pmap_common.asidCnt = newAsid;
 	}
 
-	pmap_common.asid_map[pmap_common.asidptr] = pmap;
-	pmap->asid_ix = pmap_common.asidptr;
+	if (pmap_common.asidMaps[newAsid] != NULL) {
```

Comment:
We could call `_pmap_asidDealloc` here to avoid code repetition

Reply:
Yeah, I agree with you. Changed.